usaspending_api/search/v2/views/spending_by_subaward_grouped.py

```diff
@@ -0,0 +1,145 @@
+        )
+        filter_options["time_period_obj"] = time_period_obj
+
+        query_with_filters = QueryWithFilters(QueryType.SUBAWARDS)
+        filter_query = query_with_filters.generate_elasticsearch_query(self.filters, **filter_options)
+        results = self.build_elasticsearch_search_with_aggregation(filter_query)
+
+        return Response(self.construct_es_response(results))
+
+    def validate_request_data(self, request_data):
+        spending_by_subaward_grouped_models = [
+            {"name": "limits", "key": "limit", "type": "integer", "default": 10},
+            {"name": "ordered", "key": "order", "type": "text", "text_type": "search", "default": "desc"},
+            {
+                "name": "object_class",
+                "key": "filters|object class",
+                "type": "array",
+                "array_type": "integer",
+                "text_type": "search",
+                "array_max": maxsize,
+            },
+            {
+                "name": "sorted",
+                "key": "sort",
+                "type": "text",
+                "text_type": "search",
+                "default": "award_id",
+            },
+        ]
+
+        # Accepts the same filters as spending_by_award
+        spending_by_subaward_grouped_models.extend(copy.deepcopy(AWARD_FILTER_NO_RECIPIENT_ID))
+        spending_by_subaward_grouped_models.extend(copy.deepcopy(PAGINATION))
+
+        tiny_shield = TinyShield(spending_by_subaward_grouped_models)
+        return tiny_shield.block(request_data), spending_by_subaward_grouped_models
+
+    def construct_es_response(self, results: list[subaward_grouped_model]):
+        return {
+            "limit": self.pagination["limit"],
+            "results": results,
+            "page_metadata": {
+                "page": self.pagination["page"],
+                "hasNext": True if len(results) > self.pagination["limit"] else False,
+            },
+            "messages": get_generic_filters_message(
+                self.original_filters.keys(), [elem["name"] for elem in self.models]
+            ),
+        }
+
+    def build_elasticsearch_search_with_aggregation(self, filter_query: ES_Q):
+        # Aggregate the ES query to group the subaward values by their prime award
+        terms_aggregation = A("terms", field="award_piid_fain")
+
+        # Sum the subaward amount within each prime award
+        terms_aggregation.metric("subaward_obligation", "sum", field="subaward_amount")
+
+        search_sum = SubawardSearch().filter(filter_query)
+        search_sum.aggs.bucket("award_id", terms_aggregation)
+        response = search_sum.handle_execute()
+
+        if response is None:
+            raise Exception("Breaking generator, unable to reach cluster")
+
+        results = []
+        count = 0
+        for result in response["aggregations"]["award_id"]["buckets"]:
+            award_generated_internal_id = response["hits"]["hits"][count]["_source"]["unique_award_key"]
```

Comment:
Instead of defining a `count` variable and iterating it, you can use python's `enumerate` to get the count this way:
```suggestion
        for i, result in enumerate(response["aggregations"]["award_id"]["buckets"]):
            award_generated_internal_id = response["hits"]["hits"][i]["_source"]["unique_award_key"]
```
Can we also check to make sure that the `award_piid_fain` in `result.key` matches the value on the hit to ensure that we're getting the right `unique_award_key`?  Relying on this list of hits always being in the right order seems like it could be a little fragile.  Alternatively, this could be a nested `agg` by adding another `.metric("unique_award_key", "terms", field="unique_award_key")` and referenced like ` result.unique_award_key.buckets[0].key`